plugin-gestures/src/main/java/com/mapbox/maps/plugin/gestures/GesturesPluginImpl.kt

```diff
@@ -1047,8 +1048,9 @@ class GesturesPluginImpl : GesturesPlugin, GesturesSettingsBase, MapStyleObserve
       pitch - (SHOVE_PIXEL_CHANGE_FACTOR * (deltaPixelsSinceLast + deferredShove))
     deferredShove = 0.0
     pitch = clamp(optimizedPitch, MINIMUM_PITCH, MAXIMUM_PITCH)
+    // changing pitch always should happen relative to the screen center otherwise camera may fly away
     easeToImmediately(
-      CameraOptions.Builder().pitch(pitch).build(),
+      CameraOptions.Builder().anchor(centerScreen).pitch(pitch).build(),
```

Comment:
Isn't it reproduced for other gestures?

Reply:
all other gestures (except move that actually gets camera options from the core) set an actual anchor